crates/core/src/polynomial/arith_circuit.rs

```diff
@@ -536,6 +531,32 @@ fn apply_binary_op<F: Field, P: PackedField<Scalar: ExtensionField<F>>>(
 	}
 }
 
+fn alloc_scratch_space<T, U, F>(size: usize, callback: F) -> U
+where
+	F: FnOnce(&mut [MaybeUninit<T>]) -> U,
+{
+	#[cfg(miri)]
+	{
+		use std::mem;
+		// We don't want to deal with running destructors.
+		assert!(!mem::needs_drop::<T>());
+		let mut scratch_space = Vec::<T>::with_capacity(size);
+		unsafe {
+			scratch_space.set_len(size);
+			let out =
+				callback(mem::transmute::<&mut [T], &mut [MaybeUninit<T>]>(scratch_space.as_mut()));
+			drop(scratch_space);
+			out
+		}
+	}
+	#[cfg(not(miri))]
+	{
+		// `stackalloc_uninit` throws a debug assert if `size` is 0, so set minimum of 1.
+		let size = size.max(1);
+		stackalloc::stackalloc_uninit(size, callback)
+	}
+}
+
```

Comment:
I would rather make an in-place wrapper with the same name and put it to utils (I assume we may have the same problem in other places).
Also we may get rid of the annoying panic when length is zero

Reply:
I was not able to find other places that use `stackalloc` in this way, though, there are other places in which its used.

Now, my gut feeling says that stackalloc does not give real advantage over using `Vec`. Exploring this is actually part of the task here (pt.3).

Now with all that said, do you still  think this is really a blocking issue? And if so, could you elaborate what do you mean by an in-place wrapper?

Reply:
Yes, you're right, now this is the only case for `stackalloc_uninit`, the other ones seems to be removed from the codebase.

>Now, my gut feeling says that stackalloc does not give real advantage over using Vec
In the synthetic tests there was a significant deffirence between `stackalloc` and `Vec`/`SmallVec`. Regarding this particular place possibly we should compare for the case of the most simple polynomial, e.g. `Var(0) + Va(1)` if there is any difference.

Reply:
>And if so, could you elaborate what do you mean by an in-place wrapper

I just meant to move to the utils and rename to `stackalloc_uninit`